lib/pos_blockchain.go

```diff
@@ -87,7 +87,7 @@ func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, currentView uint6
 	}
 
 	// 4. Handle reorgs if necessary
-	if bc.shouldReorg(desoBlock) {
+	if bc.shouldReorg(desoBlock, currentView) {
```

Comment:
Rather than calling `shouldReorg` and `handleReorg` here, it's clearer to just have the `handleReorg` function perform the `shouldReorg` check internally. This way, all reorg logic happens within a single function call, and the internal checks are abstracted way from ProcessBlock.

See my refactor suggestion in https://github.com/deso-protocol/core/pull/737